cmd/stackdriver-reverse-proxy/main.go

```diff
@@ -117,12 +148,43 @@ type transport struct {
 	Trace *trace.Client
 }
 
-func (t *transport) RoundTrip(req *http.Request) (*http.Response, error) {
+func (t *transport) RoundTrip(req *http.Request) (resp *http.Response, err error) {
 	s := t.Trace.SpanFromRequest(req)
 	defer s.Finish()
 
 	req.Header.Set("X-Cloud-Trace-Context", s.Header())
-	resp, err := http.DefaultTransport.RoundTrip(req)
+
+	if metricsReporter != nil {
```

Comment:
We should make metricsReporter a field in the style of t.Trace. Or have a global trace client.

Reply:
SGTM, thanks.